test/test_cuda.py

```diff
@@ -3280,5 +3280,200 @@ class TestNamedTupleInput_1(NamedTuple):
             self.assertEqual(expected_a, x.a)
             self.assertEqual(expected_b, x.b)
 
+    def test_graph_capture_simple(self):
+        s1 = torch.cuda.Stream()
+
+        with torch.cuda.stream(s1):
+            a = torch.zeros((1000,), device="cuda")
+            a += 1
+            g = torch.cuda.Graph()
+            g.capture_begin()
+            a += 1
+            g.capture_end()
+        torch.cuda.current_stream().wait_stream(s1)
+
+        g.replay()
+        g.replay()
+
+        self.assertTrue(a.sum().item() == 3000.)
+
+    def test_graph_rng_functional(self):
+        # The caching allocator isn't yet graph-safe.
```

Comment:
Yeah, so what's the plan for this anyway? Thread local allocator override?

Reply:
https://www.youtube.com/watch?v=Lf6aMcnR9WQ
Long term (many months) I'm working with cudaMallocAsync people to ensure cudaMallocAsync plays well with cuda graphs.  Hopefully after we integrate cudaMallocAsync, capturing allocations will "just work."

Short term, Arslan Zulfiqar, one of our people, hacked the Pytorch allocator to request and reserve a stream-silo for the graph(s):  graph capture gets its own memory silo/pool in which it can reuse memory, and it won't be affected by eager allocations.  My very next task after this PR is to look into upstreaming his changes or some variation.

Reply:
So, are you saying, when cudaMallocAsync becomes a thing, we get rid of PyTorch's CUDA caching allocator?

Reply:
I think that would be a sad waste, not to mention loss of control for corner cases when you need it.  I think the perfect world would be, cudaMallocAsync and the Pytorch allocator are alternative backends for some common allocator interface (which may also accept external allocators that support the same interface https://github.com/pytorch/pytorch/issues/43144)